### PR TITLE
Convert Oauth, Polling, and Projects to TS

### PR DESCRIPTION
**Pull request by camden11**
## Description and Context
This converts the following `lib` files to TS
* `oauth`
* `polling`
* `projects`

It also restructures the projects lib files, since `lib/projects` was over 1200 lines long. All projects related lib files have been moved into the `/projects` dir. `lib/projects.ts` has been split into two files - all functionality related to builds and deploys has been moved to `lib/projects/buildAndDeploy.ts`. The remaining functionality is not in `lib/projects/index.ts`

Need to merge and release this LDL PR before type checks will pass
https://github.com/HubSpot/hubspot-local-dev-lib/pull/221

## Who to Notify
@brandenrodgers @joe-yeager @kemmerle

### CODE REVIEW

File: types/projects.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { Build, SubbuildStatus } from '@hubspot/local-dev-lib/types/Build';
```
Comment (brandenrodgers) [inline types/projects.ts:1]: Nit, but in LDL we capitalize the filenames for everything in the `types/` folder

**Comment by brandenrodgers** [comment]
Up to you, but if you wanted you could probably cut `projects/index.ts` down by another ~100 lines by creating a `projects/upload.ts` file that includes `uploadProjectFiles` and `handleProjectUpload`

File: lib/projects/index.ts [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+        tempFile.name,
+        uploadMessage,
+        projectConfig.platformVersion
+      );
+
+      if (error) {
+        console.log(error);
+        uploadResult = { uploadError: error };
+      } else if (callbackFunc) {
+        console.log('callbackfunc');
```
Comment (brandenrodgers) [inline lib/projects/index.ts:449]: Looks like a console log snuck in

**Comment by camden11** [comment]
I changed the casing of `types/Prompts` and it looks like prettier took the liberty of updating the styling of every file with that import. If it's easier to review I can try to undo those changes, but you can also just ignore every prompt file that isn't project related

**Review by brandenrodgers — APPROVED** [review]
tested locally and everything still seems to work well! :ok:

**Pull request merged**